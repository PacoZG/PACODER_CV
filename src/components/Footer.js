import React from 'react'
import { useTranslation } from 'react-i18next'
import { FacebookShareButton, FacebookIcon, LinkedinIcon, LinkedinShareButton } from 'react-share'
import { ReactComponent as Address } from '../utils/assets/address.svg'
import { ReactComponent as Email } from '../utils/assets/email.svg'

const Footer = () => {
  const { t } = useTranslation()
  return (
    <div className="relative inset-x-0 bottom-0 p-4 pt-3 pb-3 bg-blue-1000 border-t border-black dark:border-gray-200 transition duration-500">
      <div className="flex flex-col md:flex-row justify-around items-center md:items-start space-y-2 md:space-y-0 md:p-2">
        <div className="flex flex-col items-center md:items-start space-y-2">
          <div className="flex flex-col items-center md:items-start space-y-2 md:space-y-0 ">
            <div className="flex items-center space-x-1">
              <Address className="h-3 w-3 text-gray-300" />
              <p className="text-gray-300 text-sm md:pl-1">00920, Helsinki</p>
            </div>
          </div>
          <div className="flex space-x-1 items-center">
            <Email className="h-4 w-4 text-gray-300" />
            <p className="text-gray-300 text-sm space-y-2 md:space-y-0 ">pako.zavala@gmail.com</p>
          </div>
        </div>
        <div>
          <div className="flex flex-col items-center space-y-2">
            <p className="text-center text-gray-300 text-xs font-semibold">{t('Footer.share-cv')}</p>
            <div className="flex items-center justify-between space-x-4">
              <div>
                <FacebookShareButton url={window.location.href} className=" font-medium focus-within:outline-none ">
                  <FacebookIcon size={24} round={true} />
                </FacebookShareButton>
              </div>
              <div>
                <p className="text-white text-base pb-1">{' & '}</p>
              </div>
              <div>
                <LinkedinShareButton url={window.location.href} className=" font-medium focus-within:outline-none ">
                  <LinkedinIcon size={24} round={true} />
                </LinkedinShareButton>
              </div>
            </div>
          </div>
        </div>

        <div className="flex flex-col items-center md:items-start space-y-1">
          <p className="text-sm text-gray-300">{t('Footer.developed-by')}</p>
          <a
            href="https://www.linkedin.com/in/francisco-zavala"
            target="blank"
            className="hover:text-gray-400 text-gray-300 text-sm"
          >
            <div className="flex items-center space-x-1">
              <LinkedinIcon size={28} round={true} />
              <p>{' Francisco Zavala'}</p>
            </div>
          </a>
        </div>
      </div>
    </div>
  )
}

export default Footer
