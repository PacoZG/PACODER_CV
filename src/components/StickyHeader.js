import React from 'react'

const StickyHeader = ({ text, theme }) => {
  return <div className={theme}>{text}</div>
}

export default StickyHeader
