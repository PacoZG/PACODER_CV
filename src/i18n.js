import i18n from 'i18next'
import { initReactI18next } from 'react-i18next'
import LanguageDetector from 'i18next-browser-languagedetector'

import translationEN from './utils/language/en/translations.json'
import translationES from './utils/language/es/translations.json'

i18n
  .use(LanguageDetector)
  .use(initReactI18next)
  .init({
    debug: true,
    fallbackLng: 'EN',
    interpolation: {
      escapeValue: false, // not needed for react as it escapes by default
    },
    resources: {
      EN: {
        translation: translationEN,
      },
      ES: {
        translation: translationES,
      },
    },
  })

export default i18n
