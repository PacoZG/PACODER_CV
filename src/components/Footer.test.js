import React from 'react'
import { I18nextProvider } from 'react-i18next'
import { BrowserRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'
import i18n from '../utils/language/i18nextForTest'
import Footer from './Footer'
import StickyHeader from './StickyHeader'

describe('Testing Footer', () => {
  test('Snapshot of the Footer component', () => {
    const redered = renderer
      .create(
        <BrowserRouter>
          <I18nextProvider i18n={i18n}>
            <Footer />
          </I18nextProvider>
        </BrowserRouter>
      )
      .toJSON()
    expect(redered).toMatchSnapshot()
  })

  test('Snapshot of the Sticky Sticker component', () => {
    const redered = renderer
      .create(
        <BrowserRouter>
          <I18nextProvider i18n={i18n}>
            <StickyHeader />
          </I18nextProvider>
        </BrowserRouter>
      )
      .toJSON()
    expect(redered).toMatchSnapshot()
  })
})
