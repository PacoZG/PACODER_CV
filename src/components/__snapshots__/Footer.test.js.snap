// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Snapshot of the Footer component 1`] = `
<div
  className="relative flex justify-center inset-x-0 bottom-0 bg-blue-500 dark:bg-blue-900 border-0 transition duration-500 w-full"
>
  <div
    className="flex justify-center w-full"
  >
    <div
      className="hidden sm:flex flex-row items-start justify-center sm-gap-8 md:gap-12 lg:gap-16 w-full sm:w-11/12 lg:w-9/12 "
    >
      <div
        className="transform duration-500 pt-2 p-8 sm:p-4 pb-0 w-full sm:w-4/5 md:w-3/4 "
      >
        <form
          onSubmit={[Function]}
        >
          <div
            className="flex flex-col items-center justify-center gap-6 transition duration-500  bg-opacity-70 dark:bg-opacity-70 rounded-xl"
          >
            <div
              className="relative flex flex-col sm:flex-row justify-between items-center gap-3 w-full"
            >
              <input
                className="bg-opacity-10 bg-white rounded-md h-10 sm:h-12 w-full p-2 text-gray-800 dark:text-gray-400 text-md md:text-lg block border border-opacity-10 focus:outline-none focus:ring-1 focus:ring-gray-200 focus:border-transparent shadow-sm border-gray-100 placeholder-gray-700 dark:placeholder-gray-400 placeholder-opacity-50 dark:placeholder-opacity-50 dark:duration-500 transition duration-500 "
                name="name"
                onChange={[Function]}
                placeholder="Full Name"
                required={true}
                type="text"
                value=""
              />
              <div
                className="relative flex flex-col w-full"
              >
                <input
                  className="bg-opacity-10 bg-white rounded-md h-10 sm:h-12 p-2 text-gray-800 dark:text-gray-400 text-md md:text-lg block border border-opacity-10 focus:outline-none focus:ring-1 focus:ring-gray-200 focus:border-transparent shadow-sm border-gray-100 placeholder-gray-700 dark:placeholder-gray-400 placeholder-opacity-50 dark:placeholder-opacity-50 dark:duration-500 transition duration-500 "
                  name="email"
                  onChange={[Function]}
                  placeholder="Email"
                  required={true}
                  type="email"
                  value=""
                />
                <label
                  className="absolute -bottom-5 left-2 text-sm text-red-900 dark:text-red-300"
                />
              </div>
            </div>
            <textarea
              className="bg-opacity-10 bg-white rounded-md h-24 w-full p-2 text-gray-800 dark:text-gray-400 text-md md:text-lg block border border-opacity-10 focus:outline-none focus:ring-1 focus:ring-gray-200 focus:border-transparent shadow-sm border-gray-100 placeholder-gray-700 dark:placeholder-gray-400 placeholder-opacity-50 dark:placeholder-opacity-50 dark:duration-500 transition duration-500 "
              name="message"
              onChange={[Function]}
              placeholder="Message"
              required={true}
              type="text"
              value=""
            />
            <p
              className="hidden "
            >
              Message sent!, I'll get back to you as soon as possible.
            </p>
          </div>
          <div
            className="flex flex-col md:flex-row items-center"
          >
            <div
              className="flex flex-col sm:flex-row items-center justify-center w-full"
            >
              <div
                className="flex justify-center sm:justify-between pt-2 w-full"
              >
                <label
                  className="text-gray-300 text-center"
                >
                  <span
                    className="text-md lg:text-base sm:w-60 text-red-900 dark:text-red-300 transform duration-500 "
                  >
                    Characters  0/50 min
                  </span>
                </label>
              </div>
              <div
                className="flex justify-center sm:block w-56"
              >
                <div
                  className="scale-65 sm:scale-55"
                />
              </div>
            </div>
            <div
              className="flex flex-col items-center "
            >
              <button
                className="transition duration-500 inline-flex justify-center py-2 px-4 w-48 shadow-sm font-extralight rounded-md bg-black opacity-20 text-lg text-gray-300"
                disabled={true}
                type="submit"
              >
                Send
              </button>
            </div>
          </div>
        </form>
      </div>
      <div
        className="flex flex-col items-center gap-6 pt-10"
      >
        <div
          className="flex flex-col items-center justify-center gap-4"
        >
          <p
            className="text-center transition duration-500 text-gray-900 dark:text-gray-300  text-sm font-semibold"
          >
            DEVELOPED BY
          </p>
          <a
            className="hover:text-gray-400 transition duration-500 text-gray-900 dark:text-gray-300  text-md"
            href="https://www.linkedin.com/in/francisco-zavala"
            target="blank"
          >
            <div
              className="flex items-center gap-1"
            >
              <svg
                height={28}
                viewBox="0 0 64 64"
                width={28}
              >
                <circle
                  cx="32"
                  cy="32"
                  fill="#007fb1"
                  r="31"
                  style={Object {}}
                />
                <path
                  d="M20.4,44h5.4V26.6h-5.4V44z M23.1,18c-1.7,0-3.1,1.4-3.1,3.1c0,1.7,1.4,3.1,3.1,3.1 c1.7,0,3.1-1.4,3.1-3.1C26.2,19.4,24.8,18,23.1,18z M39.5,26.2c-2.6,0-4.4,1.4-5.1,2.8h-0.1v-2.4h-5.2V44h5.4v-8.6 c0-2.3,0.4-4.5,3.2-4.5c2.8,0,2.8,2.6,2.8,4.6V44H46v-9.5C46,29.8,45,26.2,39.5,26.2z"
                  fill="white"
                />
              </svg>
              <p>
                Francisco Zavala
              </p>
            </div>
          </a>
        </div>
        <div
          className="flex space-x-1 items-center"
        >
          <svg
            aria-hidden="true"
            className="h-4 w-4  transition duration-500 text-gray-900 dark:text-gray-300"
            fill="currentColor"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clipRule="evenodd"
              d="M14.243 5.757a6 6 0 10-.986 9.284 1 1 0 111.087 1.678A8 8 0 1118 10a3 3 0 01-4.8 2.401A4 4 0 1114 10a1 1 0 102 0c0-1.537-.586-3.07-1.757-4.243zM12 10a2 2 0 10-4 0 2 2 0 004 0z"
              fillRule="evenodd"
            />
          </svg>
          <p
            className="transition duration-500 text-gray-900 dark:text-gray-300 space-y-2 md:space-y-0 "
          >
            contact@pacoderzavala.com
          </p>
        </div>
        <div
          className=" hidden sm:flex flex-col items-center space-y-2"
        >
          <p
            className="text-center transition duration-500 text-gray-900 dark:text-gray-300 font-semibold"
          >
            SHARE MY WEBSITE
          </p>
          <div
            className="flex items-center justify-between gap-4"
          >
            <div>
              <button
                aria-label="facebook"
                className="react-share__ShareButton  font-medium focus-within:outline-none "
                onClick={[Function]}
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "border": "none",
                    "color": "inherit",
                    "cursor": "pointer",
                    "font": "inherit",
                    "padding": 0,
                  }
                }
              >
                <svg
                  height={28}
                  viewBox="0 0 64 64"
                  width={28}
                >
                  <circle
                    cx="32"
                    cy="32"
                    fill="#3b5998"
                    r="31"
                    style={Object {}}
                  />
                  <path
                    d="M34.1,47V33.3h4.6l0.7-5.3h-5.3v-3.4c0-1.5,0.4-2.6,2.6-2.6l2.8,0v-4.8c-0.5-0.1-2.2-0.2-4.1-0.2 c-4.1,0-6.9,2.5-6.9,7V28H24v5.3h4.6V47H34.1z"
                    fill="white"
                  />
                </svg>
              </button>
            </div>
            <div>
              <button
                aria-label="linkedin"
                className="react-share__ShareButton  font-medium focus-within:outline-none "
                onClick={[Function]}
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "border": "none",
                    "color": "inherit",
                    "cursor": "pointer",
                    "font": "inherit",
                    "padding": 0,
                  }
                }
              >
                <svg
                  height={28}
                  viewBox="0 0 64 64"
                  width={28}
                >
                  <circle
                    cx="32"
                    cy="32"
                    fill="#007fb1"
                    r="31"
                    style={Object {}}
                  />
                  <path
                    d="M20.4,44h5.4V26.6h-5.4V44z M23.1,18c-1.7,0-3.1,1.4-3.1,3.1c0,1.7,1.4,3.1,3.1,3.1 c1.7,0,3.1-1.4,3.1-3.1C26.2,19.4,24.8,18,23.1,18z M39.5,26.2c-2.6,0-4.4,1.4-5.1,2.8h-0.1v-2.4h-5.2V44h5.4v-8.6 c0-2.3,0.4-4.5,3.2-4.5c2.8,0,2.8,2.6,2.8,4.6V44H46v-9.5C46,29.8,45,26.2,39.5,26.2z"
                    fill="white"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="sm:hidden flex flex-col items-center justify-around w-full pb-6"
    >
      <div
        className="transform duration-500 pt-2 p-8 sm:p-4 pb-0 w-full sm:w-4/5 md:w-3/4 "
      >
        <form
          onSubmit={[Function]}
        >
          <div
            className="flex flex-col items-center justify-center gap-6 transition duration-500  bg-opacity-70 dark:bg-opacity-70 rounded-xl"
          >
            <div
              className="relative flex flex-col sm:flex-row justify-between items-center gap-3 w-full"
            >
              <input
                className="bg-opacity-10 bg-white rounded-md h-10 sm:h-12 w-full p-2 text-gray-800 dark:text-gray-400 text-md md:text-lg block border border-opacity-10 focus:outline-none focus:ring-1 focus:ring-gray-200 focus:border-transparent shadow-sm border-gray-100 placeholder-gray-700 dark:placeholder-gray-400 placeholder-opacity-50 dark:placeholder-opacity-50 dark:duration-500 transition duration-500 "
                name="name"
                onChange={[Function]}
                placeholder="Full Name"
                required={true}
                type="text"
                value=""
              />
              <div
                className="relative flex flex-col w-full"
              >
                <input
                  className="bg-opacity-10 bg-white rounded-md h-10 sm:h-12 p-2 text-gray-800 dark:text-gray-400 text-md md:text-lg block border border-opacity-10 focus:outline-none focus:ring-1 focus:ring-gray-200 focus:border-transparent shadow-sm border-gray-100 placeholder-gray-700 dark:placeholder-gray-400 placeholder-opacity-50 dark:placeholder-opacity-50 dark:duration-500 transition duration-500 "
                  name="email"
                  onChange={[Function]}
                  placeholder="Email"
                  required={true}
                  type="email"
                  value=""
                />
                <label
                  className="absolute -bottom-5 left-2 text-sm text-red-900 dark:text-red-300"
                />
              </div>
            </div>
            <textarea
              className="bg-opacity-10 bg-white rounded-md h-24 w-full p-2 text-gray-800 dark:text-gray-400 text-md md:text-lg block border border-opacity-10 focus:outline-none focus:ring-1 focus:ring-gray-200 focus:border-transparent shadow-sm border-gray-100 placeholder-gray-700 dark:placeholder-gray-400 placeholder-opacity-50 dark:placeholder-opacity-50 dark:duration-500 transition duration-500 "
              name="message"
              onChange={[Function]}
              placeholder="Message"
              required={true}
              type="text"
              value=""
            />
            <p
              className="hidden "
            >
              Message sent!, I'll get back to you as soon as possible.
            </p>
          </div>
          <div
            className="flex flex-col md:flex-row items-center"
          >
            <div
              className="flex flex-col sm:flex-row items-center justify-center w-full"
            >
              <div
                className="flex justify-center sm:justify-between pt-2 w-full"
              >
                <label
                  className="text-gray-300 text-center"
                >
                  <span
                    className="text-md lg:text-base sm:w-60 text-red-900 dark:text-red-300 transform duration-500 "
                  >
                    Characters  0/50 min
                  </span>
                </label>
              </div>
              <div
                className="flex justify-center sm:block w-56"
              >
                <div
                  className="scale-65 sm:scale-55"
                />
              </div>
            </div>
            <div
              className="flex flex-col items-center "
            >
              <button
                className="transition duration-500 inline-flex justify-center py-2 px-4 w-48 shadow-sm font-extralight rounded-md bg-black opacity-20 text-lg text-gray-300"
                disabled={true}
                type="submit"
              >
                Send
              </button>
            </div>
          </div>
        </form>
      </div>
      <div
        className="flex flex-col items-center gap-2 pt-6"
      >
        <div
          className="flex flex-col items-center md:items-start space-y-2"
        >
          <p
            className="text-center transition duration-500 text-gray-900 dark:text-gray-300  text-sm font-semibold"
          >
            DEVELOPED BY
          </p>
          <a
            className="hover:text-gray-400 transition duration-500 text-gray-900 dark:text-gray-300  text-md"
            href="https://www.linkedin.com/in/francisco-zavala"
            target="blank"
          >
            <div
              className="flex items-center gap-1"
            >
              <svg
                height={28}
                viewBox="0 0 64 64"
                width={28}
              >
                <circle
                  cx="32"
                  cy="32"
                  fill="#007fb1"
                  r="31"
                  style={Object {}}
                />
                <path
                  d="M20.4,44h5.4V26.6h-5.4V44z M23.1,18c-1.7,0-3.1,1.4-3.1,3.1c0,1.7,1.4,3.1,3.1,3.1 c1.7,0,3.1-1.4,3.1-3.1C26.2,19.4,24.8,18,23.1,18z M39.5,26.2c-2.6,0-4.4,1.4-5.1,2.8h-0.1v-2.4h-5.2V44h5.4v-8.6 c0-2.3,0.4-4.5,3.2-4.5c2.8,0,2.8,2.6,2.8,4.6V44H46v-9.5C46,29.8,45,26.2,39.5,26.2z"
                  fill="white"
                />
              </svg>
              <p>
                Francisco Zavala
              </p>
            </div>
          </a>
        </div>
        <div
          className="flex items-center gap-1 "
        >
          <svg
            aria-hidden="true"
            className="h-4 w-4 transition duration-500 text-gray-900 dark:text-gray-300 "
            fill="currentColor"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clipRule="evenodd"
              d="M14.243 5.757a6 6 0 10-.986 9.284 1 1 0 111.087 1.678A8 8 0 1118 10a3 3 0 01-4.8 2.401A4 4 0 1114 10a1 1 0 102 0c0-1.537-.586-3.07-1.757-4.243zM12 10a2 2 0 10-4 0 2 2 0 004 0z"
              fillRule="evenodd"
            />
          </svg>
          <p
            className=" transition duration-500 text-gray-900 dark:text-gray-300  text-md space-y-2 md:space-y-0 "
          >
            contact@pacoderzavala.com
          </p>
        </div>
        <p
          className="text-center transition duration-500 text-gray-900 dark:text-gray-300  text-sm font-semibold"
        >
          SHARE MY WEBSITE
        </p>
        <div
          className="flex items-center justify-between gap-4"
        >
          <div>
            <button
              aria-label="facebook"
              className="react-share__ShareButton  font-medium focus-within:outline-none "
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "border": "none",
                  "color": "inherit",
                  "cursor": "pointer",
                  "font": "inherit",
                  "padding": 0,
                }
              }
            >
              <svg
                height={28}
                viewBox="0 0 64 64"
                width={28}
              >
                <circle
                  cx="32"
                  cy="32"
                  fill="#3b5998"
                  r="31"
                  style={Object {}}
                />
                <path
                  d="M34.1,47V33.3h4.6l0.7-5.3h-5.3v-3.4c0-1.5,0.4-2.6,2.6-2.6l2.8,0v-4.8c-0.5-0.1-2.2-0.2-4.1-0.2 c-4.1,0-6.9,2.5-6.9,7V28H24v5.3h4.6V47H34.1z"
                  fill="white"
                />
              </svg>
            </button>
          </div>
          <div>
            <button
              aria-label="linkedin"
              className="react-share__ShareButton  font-medium focus-within:outline-none "
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "border": "none",
                  "color": "inherit",
                  "cursor": "pointer",
                  "font": "inherit",
                  "padding": 0,
                }
              }
            >
              <svg
                height={28}
                viewBox="0 0 64 64"
                width={28}
              >
                <circle
                  cx="32"
                  cy="32"
                  fill="#007fb1"
                  r="31"
                  style={Object {}}
                />
                <path
                  d="M20.4,44h5.4V26.6h-5.4V44z M23.1,18c-1.7,0-3.1,1.4-3.1,3.1c0,1.7,1.4,3.1,3.1,3.1 c1.7,0,3.1-1.4,3.1-3.1C26.2,19.4,24.8,18,23.1,18z M39.5,26.2c-2.6,0-4.4,1.4-5.1,2.8h-0.1v-2.4h-5.2V44h5.4v-8.6 c0-2.3,0.4-4.5,3.2-4.5c2.8,0,2.8,2.6,2.8,4.6V44H46v-9.5C46,29.8,45,26.2,39.5,26.2z"
                  fill="white"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
